src/main/java/moze_intel/projecte/integration/jei/world_transmute/WorldTransmuteEntry.java

```diff
@@ -0,0 +1,158 @@
+
+		inputFluid = fluidFromBlock(inputBlock);
+		if (inputFluid == null)
+		{
+			inputItem = itemFromBlock(inputBlock, transmutationEntry.input);
+		}
+		if (leftOutput != null)
+		{
+			leftOutputFluid = fluidFromBlock(leftOutput.getBlock());
+			if (leftOutputFluid == null)
+			{
+				leftOutputItem = itemFromBlock(leftOutput.getBlock(), leftOutput);
+			}
+		}
+		if (rightOutput != null)
+		{
+			rightOutputFluid = fluidFromBlock(rightOutput.getBlock());
+			if (rightOutputFluid == null)
+			{
+				rightOutputItem = itemFromBlock(rightOutput.getBlock(), rightOutput);
+			}
+		}
+	}
+
+	private FluidStack fluidFromBlock(Block block)
+	{
+		if (block instanceof BlockStaticLiquid)
+		{
+			if (block == Blocks.WATER)
+			{
+				return new FluidStack(FluidRegistry.WATER, 1000);
+			}
+			else if (block == Blocks.LAVA)
+			{
+				return new FluidStack(FluidRegistry.LAVA, 1000);
+			}
+		}
+		return null;
+	}
+
+	private ItemStack itemFromBlock(Block block, IBlockState state)
+	{
+		try {
+			//We don't have a world or position, but try pick block anyways
+			return block.getPickBlock(state, null, null, null, null);
+		} catch (Exception e) {
+			//It failed, probably because of the null world and pos
+			ItemStack item = new ItemStack(block);
+			int dropped = block.damageDropped(state);
+			if (item.getItem().getHasSubtypes() && dropped > 0)
+			{
+				return new ItemStack(block, 1, dropped);
+			}
+			return item;
+		}
+	}
+
+	public boolean isRenderable()
+	{
+		boolean hasInput = inputFluid != null || !inputItem.isEmpty();
+		boolean hasLeftOutput = leftOutputFluid != null || !leftOutputItem.isEmpty();
+		boolean hasRightOutput = rightOutputFluid != null || !rightOutputItem.isEmpty();
+		return hasInput && (hasLeftOutput || hasRightOutput);
+	}
+
+	@Override
+	public void getIngredients(@Nonnull IIngredients ingredients) {
+		if (inputFluid != null)
+		{
+			ingredients.setInput(FluidStack.class, inputFluid);
+		}
+		else if (!inputItem.isEmpty())
+		{
+			ingredients.setInput(ItemStack.class, inputItem);
+		}
+
+		List<FluidStack> fluidOutputs = new ArrayList<>();
+		if (leftOutputFluid != null)
+		{
+			fluidOutputs.add(leftOutputFluid);
+		}
+		if (rightOutputFluid != null)
+		{
+			fluidOutputs.add(rightOutputFluid);
+		}
+		if (!fluidOutputs.isEmpty())
+		{
+			ingredients.setOutputs(FluidStack.class, fluidOutputs);
+		}
+
+		List<ItemStack> outputList = new ArrayList<>();
+		if (!leftOutputItem.isEmpty())
+		{
+			outputList.add(leftOutputItem);
+		}
+		if (!rightOutputItem.isEmpty())
+		{
+			outputList.add(rightOutputItem);
+		}
+		if (!outputList.isEmpty())
+		{
+			ingredients.setOutputs(ItemStack.class, outputList);
+		}
+	}
+
+	@Override
+	@Nonnull
+	public java.util.List<String> getTooltipStrings(int mouseX, int mouseY)
+	{
+		if (mouseX > 67 && mouseX < 107 && mouseY > 18 && mouseY < 38)
+		{
+			return Collections.singletonList("Click in world, shift click for second output");
```

Comment:
translate

Reply:
Will do, that was part of the change I wanted to make when I editted the description this morning saying not to merge. Just by the time I got home I forgot about that change.